tests/probabilty/test_brier.py

```diff
@@ -134,3 +124,207 @@ def test_brier_doesnt_raise(fcst, obs, expected):
     # Check again but with input data as a DataSet
     result = brier_score(xr.Dataset({"x": fcst}), xr.Dataset({"x": obs}), check_args=False)
     xr.testing.assert_equal(result, xr.Dataset({"x": expected}))
+
+
+@pytest.mark.parametrize(
+    (
+        "fcst",
+        "obs",
+        "ensemble_member_dim",
+        "thresholds",
+        "preserve_dims",
+        "reduce_dims",
+        "weights",
+        "fair_correction",
+        "threshold_mode",
+        "expected",
+    ),
+    [
+        # Fair=False, single threshold, preserve all, threshold is an int
+        (
+            btd.DA_FCST_ENS,
+            btd.DA_OBS_ENS,
+            "ens_member",
+            1,
+            "all",
+            None,
+            None,
+            False,
+            ">=",
+            btd.EXP_BRIER_ENS_ALL,
+        ),
+        # Fair=True, single threshold, preserve all, threshold is a float
+        (
+            btd.DA_FCST_ENS,
+            btd.DA_OBS_ENS,
+            "ens_member",
+            1.0,
+            "all",
+            None,
+            None,
+            True,
+            ">=",
+            btd.EXP_BRIER_ENS_FAIR_ALL,
+        ),
+        # Test reduce_dim arg
+        (
+            btd.DA_FCST_ENS,
+            btd.DA_OBS_ENS,
+            "ens_member",
+            1,
+            None,
+            "stn",
+            None,
+            True,
+            ">=",
+            btd.EXP_BRIER_ENS_FAIR_ALL_MEAN,
+        ),
+        # Fair=False, multiple_thresholds, preserve all
+        (
+            btd.DA_FCST_ENS,
+            btd.DA_OBS_ENS,
+            "ens_member",
+            np.array([-100, 1, 100]),
+            "all",
+            None,
+            None,
+            False,
+            ">=",
+            btd.EXP_BRIER_ENS_ALL_MULTI,
+        ),
+        # Test with broadcast with a lead day dimension with Fair=True
+        (
+            btd.DA_FCST_ENS_LT,
+            btd.DA_OBS_ENS,
+            "ens_member",
+            1,
+            "all",
+            None,
+            None,
+            True,
+            ">=",
+            btd.EXP_BRIER_ENS_FAIR_ALL_LT,
+        ),
+        # Test with weights
+        (
+            btd.DA_FCST_ENS,
+            btd.DA_OBS_ENS,
+            "ens_member",
+            1,
+            None,
+            "stn",
+            btd.ENS_BRIER_WEIGHTS,
+            False,
+            ">=",
+            btd.EXP_BRIER_ENS_WITH_WEIGHTS,
+        ),
+        # Test with Datasets
+        (
+            btd.FCST_ENS_DS,
+            btd.OBS_ENS_DS,
+            "ens_member",
+            1,
+            "all",
+            None,
+            None,
+            True,
+            ">=",
+            btd.EXP_BRIER_ENS_FAIR_ALL_DS,
+        ),
+        # Check threshold_mode='>'
+        (
+            btd.DA_FCST_ENS,
```

Comment:
please include a test where threshold mode > also applies to converting ensemble member forecasts to binary forecasts

Reply:
Done (and added for < and <=)